nucypher/blockchain/eth/sol/source/contracts/StakingEscrow.sol

```diff
@@ -329,22 +329,28 @@ contract StakingEscrow is Issuer {
     * @notice Get the value of locked tokens for active stakers in (getCurrentPeriod() + _periods) period
     * as well as stakers and their locked tokens
     * @param _periods Amount of periods for locked tokens calculation
+    * @param _startIndex Start index for looking in stakers array
+    * @param _maxStakers Max stakers for looking, if set 0 then all will be used
     * @return allLockedTokens Sum of locked tokens for active stakers
     * @return activeStakers Array of stakers and their locked tokens. Stakers addresses stored as uint256
     **/
-    function getAllActiveStakers(uint16 _periods)
+    function getActiveStakers(uint16 _periods, uint256 _startIndex, uint256 _maxStakers)
         external view returns (uint256 allLockedTokens, uint256[2][] memory activeStakers)
     {
         require(_periods > 0);
 
         uint256 stakersLength = stakers.length;
+        require(_startIndex < stakersLength);
         activeStakers = new uint256[2][](stakersLength);
```

Comment:
Wouldn't be better to allocate the array with `stakersLength - _startIndex` length (after `stakersLength` is updated later, of course)? It'd reduce gas, which is one of the reasons we need this PR, even if it's a call.

Reply:
I don't think so, because memory array initialization - get empty memory pointer and write length variable.
But I'll try it, maybe it will reduce gas

Reply:
It gave a little bit gas savings, thanks for suggestions
✔️